packages/edit-site/src/store/utils.js

```diff
@@ -6,21 +6,30 @@ import memoize from 'memize';
 /**
  * WordPress dependencies
  */
-import { isTemplatePart } from '@wordpress/blocks';
+import { isTemplatePart, parse } from '@wordpress/blocks';
 
-const EMPTY_ARRAY = [];
+function getBlocksFromRecord( record ) {
+	if ( record?.blocks ) {
+		return record?.blocks;
+	}
+
+	return record?.content && typeof record.content !== 'function'
+		? parse( record.content )
```

Comment:
Sorry, I didn't hit submit on my review comment for this line. I think this results in a `clientId` that doesn't match the existing blocks within the template, which means that the template area buttons no longer link back to the block in the template. I'm assuming a little here, so apologies if it's not what's causing it!

https://github.com/WordPress/gutenberg/assets/14988353/1ade99a7-9e9e-426c-80ee-d503c68b3c7b

In 6.3, the buttons link to the template part:

https://github.com/WordPress/gutenberg/assets/14988353/65258b12-318d-4c28-ad36-2a0fa906c2d3

Reply:
Good catch, @andrewserong! I forgot to test the Document settings sidebar and missed the issue.

You're right. Parsing the content separately will cause the `clientId` miss-match. I will start looking for a solution.